### PR TITLE
feat: Adding invite helpers using transformer 

### PR DESCRIPTION
**Pull request by livelove1987**
Created support for invite request using helpers

### CODE REVIEW

File: packages/bot/src/handlers/invites/INVITE_CREATE.ts [hunk 1]
```diff
@@ -1,6 +1,36 @@
 import type { DiscordGatewayPayload, DiscordInviteCreate } from '@discordeno/types'
-import type { Bot } from '../../index.js'
+import type { Bot, Invite } from '../../index.js'
 
 export async function handleInviteCreate(bot: Bot, data: DiscordGatewayPayload): Promise<void> {
-  bot.events.inviteCreate?.(bot.transformers.invite(bot, data.d as DiscordInviteCreate))
+  const invite = data.d as DiscordInviteCreate
+  const transformedInvite = {
```
Comment (Skillz4Killz) [inline packages/bot/src/handlers/invites/INVITE_CREATE.ts:6]: Why is this not in a transformer?

File: packages/bot/src/transformers/invite.ts [hunk 2]
```diff
@@ -1,15 +1,16 @@
...
+import type { InviteStageInstance } from './stageInviteInstace.js'
 
 // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
-export function transformInvite(bot: Bot, payload: DiscordInviteCreate) {
+export function transformInvite(bot: Bot, payload: DiscordInviteMetadata) {
   const props = bot.transformers.desiredProperties.invite
   const invite = {} as Invite
 
-  if (props.channelId && payload.channel_id) invite.channelId = bot.transformers.snowflake(payload.channel_id)
+  if (props.channelId && payload.channel && payload.channel.id) invite.channelId = bot.transformers.snowflake(payload.channel.id)
```
Comment (Skillz4Killz) [inline packages/bot/src/transformers/invite.ts:10]:
```suggestion
  if (props.channelId && payload.channel?.id) invite.channelId = bot.transformers.snowflake(payload.channel.id)
```

Comment (Skillz4Killz) [reply packages/bot/src/transformers/invite.ts:10]: What happened to the channel_id prop? Is that never going to be provided anymore?

File: packages/bot/src/transformers/invite.ts [hunk 3]
```diff
@@ -1,15 +1,16 @@
...
   if (props.code && payload.code) invite.code = payload.code
   if (props.createdAt && payload.created_at) invite.createdAt = Date.parse(payload.created_at)
-  if (props.guildId && payload.guild_id) invite.guildId = bot.transformers.snowflake(payload.guild_id)
+  if (props.guildId && payload.guild && payload.guild.id) invite.guildId = bot.transformers.snowflake(payload.guild?.id)
```
Comment (Skillz4Killz) [inline packages/bot/src/transformers/invite.ts:13]: Same two things as above

File: packages/bot/src/transformers/invite.ts [hunk 4]
```diff
@@ -20,6 +21,24 @@ export function transformInvite(bot: Bot, payload: DiscordInviteCreate) {
...
+
+  if (props.approximatePresenceCount && payload.approximate_presence_count) {
+    invite.approximatePresenceCount = payload.approximate_presence_count
+  }
+  if (props.guildScheduledEvent && payload.guild_scheduled_event) {
+    invite.guildScheduledEvent = payload.guild_scheduled_event ? bot.transformers.scheduledEvent(bot, payload.guild_scheduled_event) : undefined
+  }
+  if (props.stageInstance && payload.guild?.id && payload.stage_instance) {
+    invite.stageInstance = payload.stage_instance
+      ? bot.transformers.inviteStageInstance(bot, { ...payload.stage_instance, guildId: <BigString>invite.guildId })
```
Comment (Skillz4Killz) [inline packages/bot/src/transformers/invite.ts:36]: whats this BigString here for?

File: packages/bot/src/transformers/invite.ts [hunk 5]
```diff
@@ -20,6 +21,24 @@ export function transformInvite(bot: Bot, payload: DiscordInviteCreate) {
...
+      : undefined
+  }
+  if (props.expiresAt) {
+    invite.expiresAt = payload.expires_at ? Date.parse(payload.expires_at) : undefined
```
Comment (Skillz4Killz) [inline packages/bot/src/transformers/invite.ts:40]: the payload.expires_at should be checked in the condition and then this ternary is not needed

File: packages/bot/src/transformers/invite.ts [hunk 6]
```diff
@@ -1,15 +1,16 @@
-import type { DiscordApplication, DiscordInviteCreate } from '@discordeno/types'
-import type { Application, Bot, User } from '../index.js'
+import type { DiscordApplication, DiscordInviteMetadata } from '@discordeno/types'
+import type { Application, Bot, ScheduledEvent, User } from '../index.js'
+import type { InviteStageInstance } from './stageInviteInstace.js'
 
 // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
-export function transformInvite(bot: Bot, payload: DiscordInviteCreate) {
+export function transformInvite(bot: Bot, payload: DiscordInviteMetadata) {
```
Comment (Skillz4Killz) [inline packages/bot/src/transformers/invite.ts:6]:
```suggestion
export function transformInvite(bot: Bot, payload: DiscordInviteCreate  & Partial<DiscordInviteMetadata>) {
```

File: packages/bot/src/transformers/invite.ts [hunk 7]
```diff
@@ -1,15 +1,16 @@
...
   const props = bot.transformers.desiredProperties.invite
   const invite = {} as Invite
 
-  if (props.channelId && payload.channel_id) invite.channelId = bot.transformers.snowflake(payload.channel_id)
+  if (props.channelId && payload.channel?.id) invite.channelId = bot.transformers.snowflake(payload.channel.id)
```
Comment (Skillz4Killz) [inline packages/bot/src/transformers/invite.ts:10]: support payload.channel_id

File: packages/bot/src/transformers/stageInviteInstace.ts [hunk 8]
```diff
@@ -0,0 +1,35 @@
+import type { BigString, DiscordInviteStageInstance, DiscordMember } from '@discordeno/types'
+import type { Bot, Member } from '../index.js'
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+export function transformInviteStageInstance(bot: Bot, payload: DiscordInviteStageInstance & { guildId: BigString }) {
+  const props = bot.transformers.desiredProperties.inviteStageInstance
+  const inviteStageInstance = {} as InviteStageInstance
+
+  if (props.members) {
```
Comment (Skillz4Killz) [inline packages/bot/src/transformers/stageInviteInstace.ts:9]: forgot to check payload

File: packages/bot/src/handlers/invites/INVITE_CREATE.ts [hunk 9]
```diff
@@ -2,5 +2,6 @@ import type { DiscordGatewayPayload, DiscordInviteCreate } from '@discordeno/typ
 import type { Bot } from '../../index.js'
 
 export async function handleInviteCreate(bot: Bot, data: DiscordGatewayPayload): Promise<void> {
-  bot.events.inviteCreate?.(bot.transformers.invite(bot, data.d as DiscordInviteCreate))
+  const invite = bot.transformers.invite(bot, data.d as DiscordInviteCreate)
```
Comment (Skillz4Killz) [inline packages/bot/src/handlers/invites/INVITE_CREATE.ts:5]: Undo the change in this file

Comment (Skillz4Killz) [reply packages/bot/src/handlers/invites/INVITE_CREATE.ts:5]: its better without the need of the const

File: packages/bot/src/transformers/invite.ts [hunk 10]
```diff
@@ -1,15 +1,23 @@
...
+  DiscordInviteMetadata,
+  DiscordInviteStageInstance,
+  DiscordScheduledEvent,
+} from '@discordeno/types'
+import type { Application, Bot, ScheduledEvent, User } from '../index.js'
+import type { InviteStageInstance } from './stageInviteInstance.js'
 
 // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
-export function transformInvite(bot: Bot, payload: DiscordInviteCreate) {
+export function transformInvite(bot: Bot, payload: DiscordInviteCreate | DiscordInviteMetadata) {
```
Comment (Skillz4Killz) [inline packages/bot/src/transformers/invite.ts:12]: This is not the right way to do it. I suggested using & here. See my previous comment

File: packages/bot/src/transformers/invite.ts [hunk 11]
```diff
@@ -1,15 +1,23 @@
...
   const props = bot.transformers.desiredProperties.invite
   const invite = {} as Invite
-
-  if (props.channelId && payload.channel_id) invite.channelId = bot.transformers.snowflake(payload.channel_id)
+  if ((props.channelId && (<DiscordInviteCreate>payload).channel_id) || (<DiscordInviteMetadata>payload).channel?.id)
```
Comment (Skillz4Killz) [inline packages/bot/src/transformers/invite.ts:15]: not a fan of this at all

Comment (livelove1987) [reply packages/bot/src/transformers/invite.ts:15]: No different way to do that

File: packages/bot/src/handlers/invites/INVITE_CREATE.ts (hunk 9 above)
Comment (livelove1987) [reply packages/bot/src/handlers/invites/INVITE_CREATE.ts:5]: Okkay
